storage/tools/common.go

```diff
@@ -6,18 +6,30 @@ import (
 	"github.com/google/trillian"
 	"github.com/google/trillian/storage"
 	"github.com/google/trillian/storage/mysql"
+	"github.com/google/trillian/server"
 )
 
 var logIdFlag = flag.String("logid", "logId", "The log id to use")
 var treeIdFlag = flag.Int64("treeid", 3, "The tree id to use")
 var storageTypeFlag = flag.String("storage_type", "mysql", "Which type of storage to use")
 var mysqlUriFlag = flag.String("mysql_uri", "test:zaphod@tcp(127.0.0.1:3306)/test",
 	"uri to use with mysql storage")
+var logPortFlag = flag.Int("port", 8090, "Port to serve log requests on")
 
 func GetLogIdFromFlagsOrDie() trillian.LogID {
 	return trillian.LogID{[]byte(*logIdFlag), *treeIdFlag}
 }
 
+// TODO: This needs to be tidied up
+func GetLogStorageProviderFromFlags() server.LogStorageProviderFunc {
+	return func(x int64) *storage.LogStorage {
+		// TODO: We need to sort out exactly what a log id is. We really only need a tree id
+		// but having the log id as well could be a handy cross check as they can be guessed
+		storage := GetStorageFromFlagsOrDie(trillian.LogID{[]byte("This needs fixing"), x})
```

Comment:
Not sure if that works, was getting "x is pointer to interface, not interface" errors.